modules/core/src/main/java/org/apache/ignite/internal/worker/WorkersRegistry.java

```diff
@@ -229,6 +229,8 @@ public void setSystemWorkerBlockedTimeout(long val) {
                                     "[workerName=" + worker.name() + ", threadName=" + runner.getName() +
                                     ", blockedFor=" + heartbeatDelay / 1000 + "s]");
 
+                            U.dumpThreads(log);
```

Comment:
This has already been implemented. See `IGNITE_DUMP_THREADS_ON_FAILURE`

Reply:
Yes, but, for some unknown reasons we have the following:
`private Set<FailureType> ignoredFailureTypes =
            Collections.unmodifiableSet(EnumSet.of(SYSTEM_WORKER_BLOCKED, SYSTEM_CRITICAL_OPERATION_TIMEOUT));`

Reply:
This code may not cause the node crash.
And it's execution may happen more than once.
The goal is to write the thread dump each time we have threads blocked.